File: JAVA/gestionApp/Empresa.java
package gestionApp;

import gestionApp.personas.Cliente;
import gestionApp.servicios.Servicio;

import java.io.Serializable;
import java.util.ArrayList;

public class Empresa implements Serializable {
	private static final long serialVersionUID = 1L;

	private String id;
	private String nombre;
	private int nit;
	private String descripcion;

	private ArrayList<Cliente> clientes;
	private  ArrayList<Negocio> negociosAbiertos;
	private  ArrayList<Venta> ventas;
	private  ArrayList<Servicio> serviciosDeInteres;

	private static ArrayList<Empresa> allEmpresas = new ArrayList<>();

	private static int idCounter = 0;
    
	public Empresa(String nombre,int nit, String descripcion) {
		this.id = "0".repeat(3-String.valueOf(idCounter).length()) + String.valueOf(idCounter);
		this.nombre = nombre;
		this.nit = nit;
		this.descripcion = descripcion;
		this.clientes = new ArrayList<>();
		this.negociosAbiertos = new ArrayList<>();
		this.ventas = new ArrayList<>();
		this.serviciosDeInteres = new ArrayList<>();
		allEmpresas.add(this);
		idCounter += 1;
	}

	// getters y setters 
	public String getId(){return this.id;}

	public String getNombre() {return nombre;}
	public void setNombre(String nombre) {this.nombre = nombre;}

	public int getNit() {return nit;}
	public void setNit(int nit) {this.nit = nit;}

	public String getDescripcion() {return descripcion;}
	public void setDescripcion(String descripcion) {this.descripcion = descripcion;}

	public ArrayList<Cliente> getClientes() {return clientes;}
	public void addCliente(Cliente cliente){this.clientes.add(cliente);}

	public ArrayList<Negocio> getNegociosAbiertos() {return negociosAbiertos;}
	public void setNegociosAbiertos(Negocio nuevoCliente) {this.negociosAbiertos.add(nuevoCliente);}

	public ArrayList<Venta> getVentas() {return ventas;}
	public void setVenta(Venta nuevaVenta) {this.ventas.add(nuevaVenta);}

	public static ArrayList<Empresa> getAllEmpresas(){return allEmpresas;}
	
	//esto se creo para llamarlo desde negocio 
	public  void eliminarNeg(String id){
		for(int j = 0; j < negociosAbiertos.size(); j++) {
			int posicion = negociosAbiertos.indexOf(j);
			if(negociosAbiertos.get(j).getId()==id) {
				negociosAbiertos.remove(posicion);
			}
		}
	}
	public  String toStringEmpresas(){
		return "\tDatos del Empresa \nEmpresa: "+ nombre +
		 "\nNit de la empresa: " + nit + 
		 "\nID: " + id +
		 "\nDescripcion de la empresa: " + descripcion +
		 "\n/////////////////////////////////////////////\n";
	}

	@Override
	public  String toString(){
		return "\tDatos del Empresa \nEmpresa: "+ nombre +
		 "\nNit de la empresa: " + nit + 
		 "\nID: " + id +
		 "\nDescripcion de la empresa: " + descripcion +
		 "\nNegocios Abiertos: " + negociosAbiertos +
		 "\nVentas: " + ventas +
		 "\nServicios de Interes: " + serviciosDeInteres+
		 "\n/////////////////////////////////////////////\n";
		

		
	}


	

	//FUNCIONALIDADES

}

File: Python/gestionApp/Empresa.py
from tkinter import messagebox

from requests import get


class Empresa:
    allEmpresas = []
    clientes = []
    negociosAbiertos = []
    ventas = []
    serviciosInteres = []
    def __init__(self, nombre, nit, description):
        self.id = 0 
        self.nombre = nombre
        self.nit = nit
        self.description = description
        self.allEmpresas(self)
        
        
    def getId(self): return self.id
    
    def getNombre(self): return self.nombre
    def setNombre(self,nombre): self.nombre = nombre
    
    def getNit(self): return self.nombre
    def setNit(self,nit): self.nit = nit
    
    def getDescripcion(self): return self.description
    def setDescripcion(self,description): self.description = description
        
    @classmethod    
    def getAllEmpresas(cls): return cls.allEmpresas
    @classmethod
    def setAllEmpresas(cls,allEmpresas): cls.allEmpresas = allEmpresas
    
    @classmethod
    def getClientes(cls): return cls.clientes
    @classmethod
    def setClientes(cls,clientes): cls.clientes = clientes
    
    @classmethod
    def getNegociosAbiertos(cls): return cls.negociosAbiertos
    @classmethod
    def setNegociosAbiertos(cls,negociosAbiertos): cls.negociosAbiertos = negociosAbiertos
    
    @classmethod
    def getVentas(cls): return cls.ventas
    @classmethod
    def setVentas(cls,ventas): cls.ventas = ventas 
    
    @classmethod
    def getServiciosInteres(cls): return cls.serviciosInteres
    @classmethod
    def setServiciosInteres(cls,serviciosInteres): cls.serviciosInteres = serviciosInteres
    
    def buscarEmpresa(nit):
        for i in range(len(Empresa.getAllEmpresas())):
            if (Empresa.getAllEmpresas(Empresa.getNit(i))):
                return Empresa.getAllEmpresas(get(i))     
